Source/JavaScriptCore/yarr/YarrPattern.cpp

```diff
@@ -481,31 +481,9 @@ class CharacterClassConstructor {
             int val = matches[pos+index] - ch;
             if (!val)
                 return;
-            else if (val > 0) {
-                if (val == 1) {
-                    char32_t lo = ch;
-                    char32_t hi = ch + 1;
-                    matches.remove(pos + index);
-                    if (pos + index > 0 && matches[pos + index - 1] == ch - 1) {
-                        lo = ch - 1;
-                        matches.remove(pos + index - 1);
-                    }
-                    addSortedRange(isASCII(ch) ? m_ranges : m_rangesUnicode, lo, hi);
-                    return;
-                }
+            else if (val > 0)
```

Comment:
I can't remember why I did this coalescing while adding individual characters, but I seem to remember there were pathological cases where some ops on large character classes became unwieldy, i.e. large vectors.  That might have been before I implemented the operations with chunked bitmaps.  Certainly, `coalesceTables()` does the coalescing we need before returning the created character class.

Reply:
Having more pairs pessimized a few character class matches before, but I think this is pretty niche (it can only happen when one of these pairs would cross a set boundary, not sure it actually really matters outside of specific examples. I'll revert this, we don't need to overfit.